pkg/controllers/staticroute/staticroute_controller.go

```diff
@@ -0,0 +1,285 @@
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateFailTotal, MetricResType)
+}
+
+func updateSuccess(r *StaticRouteReconciler, c *context.Context, o *v1alpha1.StaticRoute) {
+	r.setStaticRouteReadyStatusTrue(c, o)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, common.MetricResTypeStaticRoute)
+}
+
+func deleteSuccess(r *StaticRouteReconciler, _ *context.Context, _ *v1alpha1.StaticRoute) {
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteSuccessTotal, common.MetricResTypeStaticRoute)
+}
+
+func (r *StaticRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	obj := &v1alpha1.StaticRoute{}
+	log.Info("reconciling staticroute CR", "staticroute", req.NamespacedName)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeStaticRoute)
+
+	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
+		log.Error(err, "unable to fetch static route CR", "req", req.NamespacedName)
+		return common.ResultNormal, client.IgnoreNotFound(err)
+	}
+
+	// Since StaticRoute service can only be activated from NSX 4.1.0 onwards,
+	// So need to check NSX version before starting StaticRoute reconcile
+	if !r.Service.NSXClient.NSXCheckVersionForStaticRoute() {
+		err := errors.New("NSX version check failed, StaticRoute feature is not supported")
+		updateFail(r, &ctx, obj, &err)
+		// if NSX version check fails, it will be put back to reconcile queue and be reconciled after 5 minutes
+		return ResultRequeueAfter5mins, nil
+	}
+
+	if obj.ObjectMeta.DeletionTimestamp.IsZero() {
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeStaticRoute)
+		if !controllerutil.ContainsFinalizer(obj, commonservice.StaticRouteFinalizerName) {
+			controllerutil.AddFinalizer(obj, commonservice.StaticRouteFinalizerName)
+			if err := r.Client.Update(ctx, obj); err != nil {
+				log.Error(err, "add finalizer", "staticroute", req.NamespacedName)
+				updateFail(r, &ctx, obj, &err)
+				return common.ResultRequeue, err
+			}
+			log.V(1).Info("added finalizer on staticroute CR", "staticroute", req.NamespacedName)
+		}
+
+		if err := r.Service.CreateorUpdateStaticRoute(obj); err != nil {
+			log.Error(err, "operate failed, would retry exponentially", "staticroute", req.NamespacedName)
+			updateFail(r, &ctx, obj, &err)
+			// TODO: if error is not retriable, not requeue
+			return common.ResultRequeue, err
+		}
+		updateSuccess(r, &ctx, obj)
+	} else {
+		if controllerutil.ContainsFinalizer(obj, commonservice.StaticRouteFinalizerName) {
+			metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteTotal, common.MetricResTypeStaticRoute)
+			// TODO, update the value from 'default' to actual value， get OrgID, ProjectID, VPCID depending on obj.Namespace from vpc store
+			if err := r.Service.DeleteStaticRoute("default", "default", "default", string(obj.UID)); err != nil {
```

Comment:
for the controller, we only know the namespace and staticroute name, could we just pass namespace and staticroute name for the DeleteStaticRoute func and get the org/project/vpc info in StaticRouteService via the default VPC from namespace?